nservicebus/upgrades/nhibernate-timeouts-index.md

```diff
@@ -0,0 +1,119 @@
+
+If this log event is written to your log file then read the following guidance on how to apply corrections.
+
+
+## Potential issues
+
+Any of the following issues can be present:
+
+- table `TimeoutEntity` has a clustered primary key
+- index `TimeoutEntity_EndpointIdx` is non-clustered
+- index `TimeoutEntity_EndpointIdx` has an incorrect column order (should be Endpoint, Time)
+
+
+How to correct these depend on the database engine that you are using.
+
+
+## Resolving schema issues with Microsoft SQL Server
+
+This assumes that both the index column order and clustered index are incorrect. To resolve this we will drop all existing indexes and recreate them.
+
+WARNING: This procedure requires downtime. Make sure that all affected endpoint instances are not running or that the database is running in single-user mode. This is needed because we are dropping index that guarantees consistency.
+
+NOTE: Make sure that the correct database is selected. If you are using a custom schema name then update the dbo schema with your custom schema identifier.
+
+NOTE: Please first run this script on your testing stage first in order to verify the script works as expected to reduce downtime.
+
+```sql
+BEGIN TRAN
+
+DROP INDEX [TimeoutEntity_SagaIdIdx] ON [dbo].[TimeoutEntity];
+GO
+
+DROP INDEX [TimeoutEntity_EndpointIdx] ON [dbo].[TimeoutEntity];
+GO
+
+ALTER TABLE [dbo].[TimeoutEntity] DROP CONSTRAINT PK__TimeoutE__3214EC06D068BEFC
+GO
+
+ALTER TABLE [dbo].[TimeoutEntity] ADD CONSTRAINT PK__TimeoutE__3214EC06D068BEFC PRIMARY KEY NONCLUSTERED (Id)
+GO
+
+CREATE NONCLUSTERED INDEX [TimeoutEntity_SagaIdIdx] ON [dbo].[TimeoutEntity]([SagaId] ASC);
+GO
+
+CREATE CLUSTERED INDEX [TimeoutEntity_EndpointIdx] ON [dbo].[TimeoutEntity]([Endpoint] ASC, [Time] ASC);
+GO
+
+COMMIT TRAN
+```
+
+If you get any of the following messages then your current database is incorrect or you are using a custom schema. Please update the schema in the SQL statements.
+
+- Cannot drop the index 'dbo.TimeoutEntity.TimeoutEntity_SagaIdIdx', because it does not exist or you do not have permission.
+- Cannot drop the index 'dbo.TimeoutEntity.TimeoutEntity_EndpointIdx', because it does not exist or you do not have permission.
+- Cannot find the object "dbo.TimeoutEntity" because it does not exist or you do not have permissions.
+
+
+## Resolving incorrect index definition on Oracle
+
+Incorrect index definition on Oracle only applies to column order. An existing `TIMEOUTENTITY_ENDPOINTIDX` index has to be dropped and a new index with correct column order needs to be created:
+
+NOTE: This procedure does not require downtime. It is still advisable to execute it when endpoint is not under heavy load.
+
+NOTE: Please first run this script on your testing or staging environment first to verify that it works correctly.
+
+```sql
+DROP INDEX TIMEOUTENTITY_ENDPOINTIDX;
```

Comment:
@MarcinHoppe Is everything in Oracle upper case by default? Also, I don't see schemas, is that intentional?